pkg/liquidity-mining/contracts/gauges/L2GaugeCheckpointer.sol

```diff
@@ -0,0 +1,239 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IAuthorizerAdaptor.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeAdder.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IGaugeController.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IL2GaugeCheckpointer.sol";
+import "@balancer-labs/v2-interfaces/contracts/liquidity-mining/IStakelessGauge.sol";
+
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/Address.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/EnumerableSet.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/openzeppelin/ReentrancyGuard.sol";
+
+import "../admin/GaugeAdder.sol";
+import "./arbitrum/ArbitrumRootGauge.sol";
+
+/**
+ * @title L2 Gauge Checkpointer
+ * @notice Implements IL2GaugeCheckpointer; refer to it for API documentation.
+ */
+contract L2GaugeCheckpointer is IL2GaugeCheckpointer, ReentrancyGuard {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    mapping(IGaugeAdder.GaugeType => EnumerableSet.AddressSet) private _gauges;
+    IAuthorizerAdaptor private immutable _authorizerAdaptor;
+    IGaugeController private immutable _gaugeController;
+    IGaugeAdder private immutable _gaugeAdder;
+
+    constructor(GaugeAdder gaugeAdder) {
+        _gaugeAdder = gaugeAdder;
+        _gaugeController = gaugeAdder.getGaugeController();
+        _authorizerAdaptor = gaugeAdder.getAuthorizerAdaptor();
+    }
+
+    modifier withValidGaugeType(IGaugeAdder.GaugeType gaugeType) {
+        require(_isValidGaugeType(gaugeType), "Unsupported gauge type");
```

Comment:
The error message is actually correct, this should be called `withSupportedGaugeType` - the other types are not invalid, we just don't support them.

Reply:
+1, changed to `withSupportedGaugeType` and `isSupportedGaugeType`.